editoast/src/views/path/properties.rs

```diff
@@ -175,15 +237,53 @@ async fn post(
 ) -> Result<Json<PathProperties>> {
     // Extract information from parameters
     let conn = &mut db_pool.get().await?;
-    let infra_version = retrieve_infra_version(conn, infra_id).await?;
     let query_props: Properties = props.into();
     let mut valkey_conn = valkey.get_connection().await?;
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || PathfindingError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+
+    // Get track section offsets.
+    let track_section_offsets = path_properties_input.find_track_section_offset();
+
+    // Get the IDs of the track sections.
+    let object_ids = path_properties_input
+        .find_track_section_offset()
+        .iter()
+        .map(|ts| ts.0.clone()) // Extract track section IDs.
+        .collect();
+
+    // Fetch the track section objects using the IDs.
+    let objects = infra
+        .get_objects(conn, ObjectType::TrackSection, &object_ids)
+        .await?;
+
+    // Prepare a vector to store position ratios.
+    let mut path_item_position_ratio = Vec::new();
+
+    for object in objects {
+        let track_section =
+            serde_json::from_str::<TrackSection>(object.railjson.to_string().as_str())?;
+
+        // Get the position of the track section from the offsets.
+        let position = track_section_offsets[track_section.id.as_ref()] as f64;
+
+        // Calculate the geographic length of the track section.
+        let geo_length = track_section.geo_bbox().diagonal_length();
+
+        // Calculate the position ratio within the track section.
+        let geo_position = position * geo_length / (track_section.length * 1000_f64);
+        let ratio = geo_position / geo_length;
```

Comment:
If we multiply by `geo_length` then divide by it, aren't these 2 lines equivalent to

```suggestion
        let ratio = position / (track_section.length * 1000_f64);
```

Reply:
Done.